mlir/test/Conversion/MIGraphXToTosa/migraphx-to-rock-conv-nhwc.mlir

```diff
@@ -1,9 +1,9 @@
 // RUN: rocmlir-driver -kernel-pipeline migraphx,highlevel %s | FileCheck %s
 
 // CHECK-LABEL: @convNHWC
-// CHECK-SAME: memref<1x5x5x4xf32>
-// CHECK-SAME: memref<7x3x3x4xf32>
-// CHECK-SAME: memref<1x3x3x7xf32>
+// CHECK-SAME: memref<100xf32>
+// CHECK-SAME: memref<252xf32>
+// CHECK-SAME: memref<63xf32>
```

Comment:
Why do we have to change the function signature ?
Why cant we collapsed the shape as the first thing we do and use

Reply:
Because all the views are eventually written away.

The entire *point* is that the function signature - the ultimate source of truth for memrefs - is a 1D buffer.

Reply:
And the reason to do it in migraphx-to-tosa is that it makes stuff like the clone verifier behave reasonably, from where I'm standing